### PR TITLE
Improve documentation and tutorials

### PR DESCRIPTION
**Pull request by Dan-Staff**
Cleaner more documented tutorial reordered to improve understanding.
Additional diagrams.
Better visualisation of the results.
Sphinx documentation

### CODE REVIEW

**Comment by danjust** [comment]
- Short intro for the README (General concept of fornax do, why, ...)
- I've run into a few conflicts in the environment.yml - I can have a closer look on Monday
- For me, from the README it doesn't become very clear how the target graph has to be defined
- Explain the quick start results - Do you want to explain parameters like hopping distance, iterations, ...?

**Comment by Dan-Staff** [comment]
> * Short intro for the README (General concept of fornax do, why, ...)

Hi, yes. The README is a placeholder at the moment mainly for internal benefit. Can you review the tutorial?

https://github.com/CDECatapult/fornax/blob/documentation/notebooks/tutorial/Tutorial%202%20-%20Making%20a%20Query.ipynb? 

Then I will transfer that narrative to the README file if appropriate.

> * I've run into a few conflicts in the environment.yml - I can have a closer look on Monday

Thanks, that's very helpful

> * For me, from the README it doesn't become very clear how the target graph has to be defined

See point 1 above 

> * Explain the quick start results - Do you want to explain parameters like hopping distance, iterations, ...?

See point 1 above

Thanks for your help @danjust

**Comment by danjust** [comment]
Tutorial 2 looks great! Only 2 comments (apart from a general intro): 
- The transition to graph search before "Creating a target graph" looks a bit lost, in particular if the reader doesn't already know where this is going. I would emphasise this concept a bit more at this point.
- And at least a very high level explanation of the cost/score of fornex would be interesting

**Comment by danjust** [comment]
Also Tutorial 2: iron_man.svg currently doesn't exist

**Comment by Dan-Staff** [comment]
> Also Tutorial 2: iron_man.svg currently doesn't exist

Can you double check that it's there now?

**Comment by Dan-Staff** [comment]
I think there's now enough documentation to start working on it in a more iterate way

**Comment by Dan-Staff** [comment]
@mattdean-digicatapult @danjust it would be very helpful if somebody could try building the docs on their machine

also I'd really appreciate some proof reading

make sure that you tell somebody if you're going to push changed to a Jupiter notebook as they are a pain to merge.

File: README.md [hunk 1]
```diff
@@ -32,148 +32,40 @@ The available options for installing SciPy packages are listed [here](https://sc
...
-        (query_node_id, target_node_id, weight)
-        for query_node_id, target_node_id, weight
-        in string_similarities
-    ]
+### Build the Docs (requires dev dependencies)
 
-    match_starts, match_ends, weights = zip(*matches)
+```bash
+cd docs
+make html
```
Comment (mattdean-digicatapult) [inline README.md]: Needs to install `fornax` before this will work

**Comment by mattdean-digicatapult** [comment]
There are 388 unique characters in the dataset

Should be 399

**Comment by mattdean-digicatapult** [comment]
The copyright at the bottom of each page also says `Daniel Staff` but should probably say something like `Digital Catapult`

File: README.md (hunk 1 above)
Comment (Dan-Staff) [reply README.md]: I've updated the README. Commit 4bf4159

**Comment by Dan-Staff** [comment]
notebook is broken, I'm trying to fix it

**Pull request merged**